Readme on old features and rewording of title vs direntry

mossroy opened the pull request:
Getting rid of this "Title" concept modifies a lot of code.
@sharun-s : is that what you expected in #218 ?

sharun-s commented on www/js/lib/zimfile.js:
```diff
@@ -20,7 +20,7 @@
  */
 'use strict';
-define(['xzdec_wrapper', 'util', 'utf8', 'q'], function(xz, util, utf8, Q) {
+define(['xzdec_wrapper', 'util', 'utf8', 'q', 'zimDirEntry'], function(xz, util, utf8, Q, zimDirEntry) {
```
Not sure how to add a comment on the "blob" function. So am adding here.
This function took me the longest to figure out. Okay I didn't know what Promises were so that didn't help :)
But even after...I think what was missing to me was how the decompression process works. Just a brief note in the function documentation would be really useful. Like what is chunk size? Relationship to the compressed data read length. To read the 16 bytes to get blob address for example why is chunk size set to 5 KB. Basically some kind of note on how this works would be great. By changing these sizes (data length/chunk size) can reads be done faster?

sharun-s replied:
nextCluster = readInt(clusterOffsets, 8, 8); This nextCluster doesn't seem to get used anywhere

mossroy replied:
I don't master this code portion (I did not write it myself).
I agree with you that it lacks some comments. And it might be a place where some optimization could take place.
I'll leave it as it is for now.

sharun-s commented on www/js/lib/zimArchive.js:
```diff
@@ -238,12 +222,12 @@ define(['zimfile', 'zimDirEntry', 'util', 'utf8'],
-        return title.readData().then(function(data) {
-            callback(title.name(), utf8.parse(data));
+    ZIMArchive.prototype.readArticle = function(dirEntry, callback) {
+        return dirEntry.readData().then(function(data) {
```
Is this return here required? The callback handles the end of the read.

mossroy replied:
True : it's unnecessary

mossroy commented on www/js/lib/util.js:
```diff
@@ -78,16 +78,16 @@ define(['q'], function(q) {
+    function removeDuplicateTitlesInDirEntryArray(array) {
+        array.sort(function(dirEntryA, dirEntryB) {
+            if (dirEntryA.title < dirEntryB.title) return -1;
+            if (dirEntryA.title > dirEntryB.title) return 1;
```
I agree. It was probably meaningful with Evopedia archives, but is not any more.

mossroy commented on www/js/lib/zimArchive.js:
```diff
@@ -254,12 +238,12 @@ define(['zimfile', 'zimDirEntry', 'util', 'utf8'],
-            callback(title.name(), data);
+    ZIMArchive.prototype.readBinaryFile = function(dirEntry, callback) {
+        return dirEntry.readData().then(function(data) {
+            callback(dirEntry.name(), data);
```
Not quite : it hides the concept of cluster

mossroy commented:
I pushed another commit that addresses some of your comments.
I'd like to merge this PR, test and release a version 2.1.0 that would include all the recent commits we made.
Feel free to open other issues if necessary, for a next version

sharun-s commented:
Ok. Sounds good to me!
